pkg/validate/validate_component_test.go

```diff
@@ -60,107 +24,62 @@ components:
-		components   string
-		errSubstring string
-	}{
-		{
-			desc:       "success",
-			set:        defaultComponentSet,
-			components: defaultComponentData,
-			// no errors
-		},
-
-		// Tests for component sets
-		{
-			desc: "component set fail: bad kind",
-			set: `
+var defaultComponentSet = `
 apiVersion: 'bundle.gke.io/v1alpha1'
-kind: Zor
-spec:
-  setName: zip
-  version: 1.0.2
-  components:
-  - componentName: foo-comp
-    version: 1.0.2`,
-			components:   defaultComponentData,
-			errSubstring: "must be ComponentSet",
-		},
-
-		{
-			desc: "component set fail: apiVersion",
-			set: `
-apiVersion: 'zork.gke.io/v1alpha1'
 kind: ComponentSet
 spec:
-  setName: zip
+  setName: foo-set
   version: 1.0.2
   components:
   - componentName: foo-comp
-    version: 1.0.2`,
-			components:   defaultComponentData,
-			errSubstring: "bundle.gke.io/<version>",
-		},
-		{
-			desc: "component set fail: invalid X.Y.Z version string",
-			set: `
-apiVersion: 'bundle.gke.io/v1alpha1'
-kind: ComponentSet
-spec:
-  setName: zip
-  version: foo
-  components:
-  - componentName: foo-comp
-    version: 1.0.2`,
-			components:   defaultComponentData,
-			errSubstring: "must be of the form X.Y.Z",
-		},
-		{
-			desc: "fail: missing X.Y.Z version string",
-			set: `
-apiVersion: 'bundle.gke.io/v1alpha1'
-kind: ComponentSet
+    version: 1.0.2
+`
+
+// Component Tests
+func TestComponent(t *testing.T){
+  defaultComponent := `
+apiVersion: bundle.gke.io/v1alpha1
+kind: Component
+metadata:
+  creationTimestamp: null
 spec:
-  setName: zip
-  components:
-  - componentName: foo-comp
-    version: 1.0.2`,
-			components:   defaultComponentData,
-			errSubstring: "Required value",
-		},
-		{
-			desc: "fail: missing set name",
-			set: `
+  version: 30.0.2
+  componentName: etcd-component`
+
+	
+	component, err := converter.FromYAMLString(defaultComponent).ToComponent()
+	if err != nil {
+		t.Fatalf("invalid parsing: %v", err)
+	}
+	errs := Component(component)
+	if len(errs) != 0 {
+		t.Fatalf("%v", errs)
+	}
+}
+
+func TestComponentNoRefs(t *testing.T){
+	defaultComponentNoRefs := `
 apiVersion: 'bundle.gke.io/v1alpha1'
-kind: ComponentSet
+kind: Component
+metadata:
+  name: foo-comp-1.0.2
 spec:
-  version: 1.0.2
-  components:
-  - componentName: foo-comp
-    version: 1.0.2`,
-			components:   defaultComponentData,
-			errSubstring: "Required value",
-		},
+  componentName: foo-comp
+  version: 2.10.1
+  appVersion: 3.10.1`
 
-		// Tests for Components
-		{
-			desc: "success",
-			set:  defaultComponentSetNoRefs,
-			components: `
-components:
-- apiVersion: 'bundle.gke.io/v1alpha1'
-  kind: Component
-  metadata:
-    name: foo-comp-1.0.2
-  spec:
-    componentName: foo-comp
-    version: 2.10.1
-    appVersion: 3.10.1`,
-		},
-		{
+    component, errConv := converter.FromYAMLString(defaultComponentNoRefs).ToComponent()
+    if errConv != nil {
+    	t.Fatalf("error converting ", errConv)
+    	return
+    }
+    errs := Component(component)
```

Comment:
Your formatting is funny. Can you make sure you have gofmt auto-enabled on write or similar?

Reply:
Still working on this.  Just pushed and it got updated.  Thought it would only update my branch and not show here.  I'll close the PR and reopen when I'm ready for review. 

I didn't intend to merge this as is.